### DISCUSSION TITLE
Classic mod for osu!taiko should receive separate difficulty calculation.

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#18787] [Q&A]
Note: this may seem very similar to #13047, #14798, and #17265. However, this discussion does not cover the differences between stable and Classic, rather how to compensate for the difficulty differences due to the two separate systems.

I'm starting this discussion in response to a few new changes, in addition to many older ones that have gone unaddressed. However, I aim to shed light on an aspect that has not been discussed anywhere that I can find, including many of the discussions I will be referencing.
In [#13047](https://github.com/ppy/osu/discussions/13047#discussioncomment-803427), chayleaf writes:
> you can achieve [4:3] without changing anything in-game by simply closing part of the screen with a towel or something else. That's why I think it's fine to allow such resolutions, even if it is unfair - simply because it can be easily achieved even without such a feature.

This is a fair argument. This would mean that all aspect ratios should be considered the same difficulty, as you can replicate them without changing any in-game settings. The same cannot be said for the Hidden mod, however. The implementation without the Classic mod has the notes fade out before the 4:3 point. You cannot replicate a 4:3 HD experience in lazer because covering the extra part of the screen will inhibit your reading ability. **This means that HD without Classic is inherently more difficult than HD with Classic.**

Please note my argument here: I am NOT saying that gameplay with or without Classic is inherently better or worse (#13047). I am NOT talking about how the current implementation of the Classic mod does not perfectly replicate stable behavior (#14798 and #17265). What I am saying is that the Classic mod should receive a different star difficulty due to the differences in gameplay that it creates.

After discussing with many community members, the intended differences in gameplay become more apparent. HR in stable is easier with widescreen resolutions due to being able to see more notes. FL in stable is easier due to the positioning of the view area. Depending on the conclusion drawn from #14798, the Classic mod may have varying degrees of differing difficulty.
> Now that `TaikoModClassic` is starting to provide more than one change in mechanisms here, it may be better defining toggle settings similar to `OsuModClassic`. Maybe in a follow-up PR.

[This quote](https://github.com/ppy/osu/pull/18764#discussion_r901150332) is from sw1tchbl4d3r's recently opened PR aiming to start bringing the Classic mod closer to stable gameplay. This would mean that options such as `Drumrolls and swells do not influence combo`, `Classic flashlight view area`, and `Lock playfield aspect ratio` might be options that would be toggle-able under mod customization. I understand the reasoning behind this increased customizability; however, this isn't considered desired behavior for many in the taiko community.

To my understanding, the Classic mod is supposed to represent stable gameplay as much as possible. If leaderboards are to be combined (which to my knowledge has not been determined), plays set in stable will have the Classic mod retroactively applied. However, these scores would not migrate intact; they would have to be recalculated to match the lazer scoring system, even though neither the Standardised nor the Classic scoring systems match the scoring seen in stable. Additionally, as noted in #9087, the accuracy of all plays will drop due to "greats" being worth 33% rather than 50%. By allowing the forced aspect ratio to be toggle-able, many may consider the Classic mod not to be a valid indicator of older/newer scores.

By not acknowledging the difficulty differences between gameplay with and without the Classic mod, players will play with the easier mod combination to receive the same rewards. Additionally, plays set in stable may not be comparable to those in lazer due to the high skill difference required. Overall, there needs to be more transparency as to how the Classic mod will be handled in the taiko ruleset, as there are many differences that still need addressing.

**Comment by Lawtrohux** [comment]
> Classic mod should receive a different star difficulty due to the differences in gameplay that it creates.

This is not in an immediate list of things to do right now and is of low priority in the active rework as of now.

> By not acknowledging the difficulty differences between gameplay with and without the Classic mod, players will play with the easier mod combination to receive the same rewards

This has been brought to attention of the taiko pp committee by multiple members of the community, and has been discussed in the past by other contributors on the osu!dev server. 

>  Overall, there needs to be more transparency as to how the Classic mod will be handled in the taiko ruleset, as there are many differences that still need addressing.

However I cannot speak for how classic mod will be 'handled' only that the differences will be recognised within future difficulty and performance developments by the committee, however i cannot speak for any other contributors.

In regards to the reading skill, a proposed rework is within the osu!taiko rework general document linked on the osu! wiki [here](https://docs.google.com/document/d/1Z5GC4DMqOVzeIERMSK3qpQaqjq-sVnhbuoxAwy9qxDs/edit).

**Comment by bdach** [comment]
This discussion is in the "too obvious to exist" category for me. Diffcalc can already handle classic vs non-classic perfectly fine. (see: osu! max combo changes). Only thing left is to define and implement the actual algorithm which I'm certainly not going to attempt to do.
